pyramid/tweens.py

```diff
@@ -42,7 +42,7 @@ def excview_tween(request):
             provides = providedBy(exc)
             try:
                 response = _call_view(
-                    registry,
+                    request.registry,
```

Comment:
Sorry but just to clarify I think the integration test and the weakref fixes on the WeakOrderedSet are good. I'm just nitpicking these other two changes.